plugins/discovery/discovery.go

```diff
@@ -220,15 +211,30 @@ func (c *Discovery) logrusFields() logrus.Fields {
 	}
 }
 
-func processBundle(ctx context.Context, manager *plugins.Manager, factories map[string]plugins.Factory, b *bundleApi.Bundle, query string, m metrics.Metrics) (*config.Config, *pluginSet, error) {
+func processBundle(ctx context.Context, manager *plugins.Manager, factories map[string]plugins.Factory, b *bundleApi.Bundle, query, service string, m metrics.Metrics) (*pluginSet, error) {
 
 	config, err := evaluateBundle(ctx, manager.ID, manager.Info, b, query)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
+	}
+
+	// check for updates to the discovery configuration
+	if config.Discovery != nil {
+		return nil, fmt.Errorf("updates to the discovery configuration are not allowed")
+	}
+
+	client := manager.Client(service)
+
+	if err := manager.Reconfigure(config); err != nil {
+		return nil, err
+	}
+
+	// check for updates to the discovery service
+	if !client.Equal(manager.Client(service)) {
+		return nil, fmt.Errorf("updates to the discovery service are not allowed")
 	}
```

Comment:
Shouldn't the check on the service config occur before we reconfigure the manager? As is, this means we'll end up with updates being applied inconsistently.

Reply:
Yes I was thinking about that which meant we need to either make `parseServicesConfig` (plugins/plugins.go) public or copy over the logic or move the function to a common package.

Any preference here ?

Reply:
A good design principle is to optimize for change--i.e., assume we're going to have to modify the code at some point. Given that, avoid copying logic because it means more maintenance and more bugs. If you're worried about increasing the API surface area outside of OPA consider exporting the function inside of an internal package.